core/modules/replica/tools/qserv-replica-application-test.cc

```diff
@@ -20,170 +20,22 @@
-         * COMMAND2 <p1>       [<o1>]                     [--verbose]
-         * 
-         * Note that the parser is guaranteed to run before invoking method "runImpl()".
-         */
-        parser().commands(
-            "command",
-            {"COMMAND1", "COMMAND2", "COMMAND3"},
-            _cmd);
-
-        parser().required(
-            "p1",
-            "description of the required parameter p1 for all commands",
-            _p1);
-
-        parser().optional(
-            "o1",
-            "description of the optional parameter o1 for all commands",
-            _o1);
-
-        parser().flag(
-            "verbose",
-            "verbose mode",
-            _verbose);
-
-        auto&& command1 = parser().command("COMMAND1");
-        command1.description(
-            "This is the first command"
-        );
-        command1.required(
-            "p11",
-            "description of the additional required parameter specific for the command",
-            _p11
-        );
-        command1.optional(
-            "o11",
-            "description of the additional optional parameter specific for the command",
-            _o11
-        );
-        command1.option(
-            "o12",
-            "description of the additional option specific to the command",
-            _o12
-        );
-
-        parser().command("COMMAND2").description(
-            "This is the second command"
-        );
-        parser().command("COMMAND2").flag(
-           "f21",
-           "description of the additional flag specific to the command",
-           _f21
-        );
-    }
-
-    /**
-     * Implement a user-defined sequence of actions here.
-     *
-     * @return completion status
-     */
-    int runImpl() final {
-        cout
-            << "Hello from TestApplication:\n"
-            << "      cmd: " << _cmd << "\n"
-            << "      p11: " << _p11 << "\n"
-            << "       p1: " << _p1  << "\n"
-            << "       p2: " << _p2  << "\n"
-            << "       o1: " << _o1  << "\n"
-            << "       o2: " << (_o2      ? "true" : "false") << "\n"
-            << "      o11: " << _o11 << "\n"
-            << "      o12: " << _o12 << "\n"
-            << "      f21: " << (_f21     ? "true" : "false") << "\n"
-            << "  verbose: " << (_verbose ? "true" : "false") << endl;
-        return 0;
-    }
-
-private:
-
-    // Values of the command line parameters
-
-    string  _cmd;
-    int     _p1;
-    string  _p11;
-    string  _p2;
-    unsigned int _o1;
-    unsigned int _o2;
-    unsigned int _o11;
-    string  _o12;
-    bool    _verbose;
-    bool    _f21;
-};
-} /// namespace
-
 int main(int argc, char* argv[]) {
     try {
-        auto const app = ::TestApplication::create(argc, argv);
+        auto const app = TestApp::create(argc, argv);
         return app->run();
     } catch (exception const& ex) {
         cerr << "main()  the application failed, exception: " << ex.what() << endl;
```

Comment:
This app is installed with qserv?

Reply:
It's not. It's installed in a separate container `qserv/replica:tools` along with other binaries starting with `qserv-replica-`.